File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = set()
    
    n = len(elements)
    elements *= 2
    
    prefix = [0] * (2 * n + 1)
    
    for i in range(2 * n):
        prefix[i + 1] = prefix[i] + elements[i]
    
    for i in range(n):
        for l in range(1, n + 1):
            answer.add(prefix[i + l] - prefix[i])
            
    return len(answer)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.kt
class Solution {
    fun solution(elements: IntArray): Int {
        val answer = mutableSetOf<Int>()
        val n = elements.size
        
        for (len in 1..n) {
            for (i in 0 until n) {
                var sum = 0
                
                for (j in i until (i + len)) {
                    sum += elements[j % n]
                }
                
                answer.add(sum)
            }   
        }
        
        return answer.size
    }
}